clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeleteExecutor.scala

```diff
@@ -103,10 +103,14 @@ class DeleteExecutor[TI <: TransferIndex](deleteCommand: DeleteCommand[TI])
   )(implicit ec: ExecutionContext): Future[UpsertId] = {
     val pathsToDelete = existingMetadata.pathsToDelete.filter { path =>
       val pathExists = ioUtil.googleObjectExists(path)
-      if (!pathExists) {
-        logger.warn(
-          s"'$path' associated with $prettyKey does not exist in the cloud."
-        )
+      val err = s"'$path' associated with $prettyKey does not exist in the cloud."
+      (pathExists, deleteCommand.forceDelete) match {
+        case (false, false) =>
+          logger.error(err + " Use --force-delete to delete this file.")
+          throw new Exception(err)
+        case (false, true) =>
+          logger.warn(err)
+        case _ =>
```

Comment:
I'd do nested `if`s instead of the `match`, to avoid the awkward no-op catch-all.